Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables

```diff
@@ -0,0 +1,684 @@
+                $name = $_.ResourceName
+                $resourceGroupName = $_.ResourceGroupName
+                $automationAccountResources += $allAutomationAccountResources | Where-Object {$_.AutomationAccountName -eq $name -and $_.ResourceGroupName -eq $resourceGroupName}
+            }
+            catch
+            {
+                Write-Host "Valid resource group name and resource name not found in input json file. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping the Resource: [$($name)]..." -ForegroundColor $([Constants]::MessageType.Warning)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Valid resource group name and resource name not found in input json file.")    
+                $logSkippedResources += $logResource
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+        # No file path provided as input to the script. Fetch all Automation Accounts in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "Fetching all Automation Accounts in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+
+            # Get all Automation Accounts in the Subscription
+            $automationAccountResources = Get-AzAutomationAccount -ErrorAction SilentlyContinue
+        }
+        else
+        {
+            if (-not (Test-Path -Path $FilePath))
+            {
+                Write-Host "Input file: [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)
+                return
+            }
+
+            Write-Host "Fetching all Automation Accounts from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $automationAccountResourcesFromFile = Import-Csv -LiteralPath $FilePath
+            $validAutomationAccountResources = $automationAccountResourcesFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.AutomationAccountName) -and ![String]::IsNullOrWhiteSpace($_.ResourceGroupName) }
+            $allAutomationAccountResources = Get-AzAutomationAccount -ErrorAction SilentlyContinue
+            $validAutomationAccountResources | ForEach-Object {
+                $resourceGroupName = $_.ResourceGroupName        
+                $name = $_.AutomationAccountName               
+
+                try
+                {
+                    $automationAccountResources += $allAutomationAccountResources | Where-Object {$_.AutomationAccountName -eq $name -and $_.ResourceGroupName -eq $resourceGroupName}
```

Comment:
same here if no matching automation account let's log/print in console.

Reply:
Addressed